File: services/maradona-java-brain/src/main/java/com/maradona/telegram/TelegramNotifier.java
package com.maradona.telegram;

import com.maradona.config.MaradonaProperties;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestClient;
import org.springframework.web.client.RestClientException;

import java.util.HashMap;
import java.util.Map;

@Service
public class TelegramNotifier {
    private final MaradonaProperties props;
    private final RestClient rest = RestClient.create();

    public TelegramNotifier(MaradonaProperties props) {
        this.props = props;
    }

    public void send(String text) {
        String botToken = props.getTelegram().getBotToken();
        String chatId = props.getTelegram().getChatId();
        boolean enabled = props.getTelegram().isTelegramEnabled();

        if (!enabled) {
            System.out.println("TELEGRAM OFF BY CONFIG telegram_enabled=false: " + text);
            return;
        }

        if (botToken == null || botToken.isBlank()) {
            System.out.println("TELEGRAM OFF: TELEGRAM_BOT_TOKEN mancante o vuoto");
            System.out.println("MESSAGGIO NON INVIATO: " + text);
            return;
        }

        if (chatId == null || chatId.isBlank()) {
            System.out.println("TELEGRAM OFF: TELEGRAM_CHAT_ID mancante o vuoto");
            System.out.println("MESSAGGIO NON INVIATO: " + text);
            return;
        }

        String url = "https://api.telegram.org/bot" + botToken + "/sendMessage";

        Map<String, Object> body = new HashMap<>();
        body.put("chat_id", chatId);
        body.put("text", text);
        body.put("disable_web_page_preview", true);

        try {
            rest.post()
                    .uri(url)
                    .contentType(MediaType.APPLICATION_JSON)
                    .body(body)
                    .retrieve()
                    .toBodilessEntity();

            System.out.println("TELEGRAM SENT OK");

        } catch (RestClientException e) {
            System.out.println("TELEGRAM ERROR: " + e.getMessage());
            System.out.println("TELEGRAM BOT TOKEN PRESENT: " + !botToken.isBlank());
            System.out.println("TELEGRAM CHAT ID: " + chatId);
            System.out.println("MESSAGGIO NON INVIATO: " + text);
        } catch (Exception e) {
            System.out.println("TELEGRAM GENERIC ERROR: " + e.getMessage());
            e.printStackTrace();
            System.out.println("MESSAGGIO NON INVIATO: " + text);
        }
    }
}

File: services/maradona-java-brain/src/telegramNotifier.js
export async function sendTelegram(text, opts = {}) {
  const token = opts.token ?? process.env.TELEGRAM_BOT_TOKEN;
  const chatId = opts.chatId ?? process.env.TELEGRAM_CHAT_ID;
  if (!token || !chatId) {
    return { ok: false, skipped: true, reason: 'TELEGRAM_BOT_TOKEN o TELEGRAM_CHAT_ID mancanti' };
  }

  const url = `https://api.telegram.org/bot${token}/sendMessage`;
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify({ chat_id: chatId, text, disable_web_page_preview: true })
  });

  const data = await res.json().catch(() => ({}));
  if (!res.ok) throw new Error(`Telegram error ${res.status}: ${JSON.stringify(data)}`);
  return data;
}
